core/src/main/kotlin/com/justai/jaicf/context/manager/InMemoryBotContextManager.kt

```diff
@@ -51,11 +51,26 @@ object InMemoryBotContextManager : BotContextManager {
 
 private fun DialogContext.clone(): DialogContext {
     val dc = DialogContext()
+
     dc.nextContext = nextContext
     dc.currentContext = currentContext
     dc.nextState = nextState
     dc.currentState = currentState
-    dc.transitions.putAll(transitions)
-    dc.backStateStack.addAll(backStateStack)
+
+    dc.transitions.apply {
+        clear()
```

Comment:
Sorry, didn't get that. Why do you need to clear an empty map of newly created `DialogContext`?

Reply:
It's not necessary for `transitions` and `backStateStack`, but it's important for `transitionHistory` as it's created not empty.
And as far it's necessary for some of the fields, I think it's better to keep this logic for all fields in case of changes in implementation details.